core/domain/skill_services.py

```diff
@@ -1147,3 +1147,74 @@ def filter_skills_by_mastery(user_id, skill_ids):
         if skill_id in filtered_skill_ids:
             arranged_filtered_skill_ids.append(skill_id)
     return arranged_filtered_skill_ids
+
+
+def get_categorized_and_untriaged_skills_dicts(
```

Comment:
Note: should only pass domain objects around -- use dicts at the server boundary only (i.e. only when communicating to the datastore and to the frontend).

Reply:
fixed now.